src/datomish_user_agent_service/api.cljc

```diff
@@ -268,26 +271,41 @@
             [(get-else $ ?save :save/title "") ?title]
             [(get-else $ ?save :save/excerpt "") ?excerpt]]))
       (map (fn [[page url title excerpt]]
-             {:url url :title title :excerpt excerpt :snippet nil :lastVisited nil})))))
+             {:url url :title title :excerpt excerpt :snippet "" :lastVisited nil})))))
 
 ;; TODO: limit, since.
 ;; TODO: return lastVisited, sort by lastVisited.
 ;; TODO: support snippet extraction.
 (defn <saved-pages-matching-string [db string]
-  (go-pair
-    (->>
-      (<?
-        (d/<q
-          db
-          {:find '[?page ?url ?title ?excerpt]
-           :in '[$]
-           :where [[(list 'fulltext '$ :any string) '[[?save]]]
-                   '[?save :save/page ?page]
-                   '[?page :page/url ?url]
-                   '[(get-else $ ?save :save/title "") ?title]
-                   '[(get-else $ ?save :save/excerpt "") ?excerpt]]}))
-      (map (fn [[page url title excerpt]]
-             {:url url :title title :excerpt excerpt :snippet nil :lastVisited nil})))))
+  (let [string (str "*" string "*")] ;; Wildcard match.
```

Comment:
This turns out to be harder than expected; I'm leaving it for now.  Real frecency will be it's own challenge.